server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestPutIndexTemplateAction.java

```diff
@@ -51,6 +60,11 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         Map<String, Object> sourceAsMap = XContentHelper.convertToMap(request.requiredContent(), false,
             request.getXContentType()).v2();
         sourceAsMap = RestCreateIndexAction.prepareMappings(sourceAsMap);
+        if (request.getRestApiVersion() == RestApiVersion.V_7 && sourceAsMap.containsKey("template")) {
```

Comment:
do we need a log message here since we already have one above ?

Reply:
Unforunately, yes. They're different scenarios -- `?template=blah` in the url versus `"template":"blah"` in the json body.